src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -708,11 +675,19 @@ public bool ApplyNullableTransforms(ImmutableArray<bool> transforms, INonNullTyp
                 result = result.WithTypeAndModifiers(newTypeSymbol, result.CustomModifiers);
             }
 
-            result = isAnnotated ?
-                result.AsNullableReferenceType() :
-                result.AsNotNullableReferenceType();
+            if (isAnnotated)
+            {
+                result = result.AsNullableReferenceType(fromDeclaration: true);
```

Comment:
>fromDeclaration: true [](start = 56, length = 21)

This is "from declaration" because we only use this method when loading from PE, correct?
If so, it may be good to comment.

Reply:
> This is "from declaration" because we only use this method when loading from PE, correct?

No this is not correct. It is also used to manipulate symbols while binding declarations. It is never used by flow analysis. BTW, we are going to get rid of ```nonNullTypesContext``` and change the ```ImmutableArray<bool> transforms``` to be an array of NullableAnnotation. So we will not need to guess or assume what value to use.

---
In reply to: [230523652](https://github.com/dotnet/roslyn/pull/30913#discussion_r230523652) [](ancestors = 230523652)